homework_3/src/main/java/com/epam/spring/service/mapper/ReceiptMapper.java

```diff
@@ -0,0 +1,14 @@
+package com.epam.spring.service.mapper;
+
+import com.epam.spring.controller.dto.ReceiptDto;
+import com.epam.spring.service.model.Receipt;
+import org.mapstruct.Mapper;
+
+@Mapper
+public interface ReceiptMapper {
+
+    ReceiptDto mapReceiptDto(Receipt receipt);
```

Comment:
Hi Viktor. After receipt creation the DTO is returned to the user with `email` field equal `null`.
 This happens since we have complex mapping from _Receipt->User->email_ to _ReceiptDto.email_ and
 `mapstruct` does not know how to handle it. Please use this construction to resolve an issue
```
@Mappings({
        @Mapping(target="email", source="receipt.user.email")
})
```
Thanks